server/services/licence/curfewAddressState.ts

```diff
@@ -73,17 +73,17 @@ const approvedPremisesAddressState = (licence) => {
 }
 
 const taskCompletion = (licence) => {
-  const { consent, electricity } = getIn(licence, ['curfew', 'curfewAddressReview']) || {}
+  const { consent, consentHavingSpoken, electricity } = getIn(licence, ['curfew', 'curfewAddressReview']) || {}
   const curfewAddress = getIn(licence, ['proposedAddress', 'curfewAddress']) || {}
   const offenderIsOccupier = getIn(curfewAddress, ['occupier', 'isOffender']) === 'Yes'
 
   if (offenderIsOccupier && electricity) {
     return TaskState.DONE
   }
-  if (consent && electricity) {
+  if ((consent && electricity) || (consentHavingSpoken && electricity)) {
     return TaskState.DONE
   }
-  if (consent || electricity) {
+  if (consent || consentHavingSpoken || electricity) {
```

Comment:
So I didn't want to introduce another 'if' as I think that's more confusing.
What the change does is is extend the circumstances as to  when the task is 'done' 
So rather than just checking against consent or  electricity, it's now additionally checking for consentHavingSpoken
so  _if (consent || electricity)_  goes to   _if (consent || consentHavingSpoken || electricity)_

Reply:
ok